subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/AvailableToolChains.java

```diff
@@ -167,24 +169,28 @@ static private ToolChainCandidate findCygwin() {
         return new UnavailableToolChain("gcc cygwin");
     }
 
-    static private ToolChainCandidate findGcc() {
+    static private List<ToolChainCandidate> findGccs() {
         GccVersionDeterminer versionDeterminer = GccVersionDeterminer.forGcc(TestFiles.execActionFactory());
 
-        List<File> gppCandidates = OperatingSystem.current().findAllInPath("g++");
-        for (int i = 0; i < gppCandidates.size(); i++) {
-            File candidate = gppCandidates.get(i);
+        Set<File> gppCandidates = new LinkedHashSet<File>(OperatingSystem.current().findAllInPath("g++"));
+        List<ToolChainCandidate> toolChains = Lists.newArrayList();
+        File firstInPath = gppCandidates.iterator().next();
+        for (File candidate : gppCandidates) {
             GccVersionResult version = versionDeterminer.getGccMetaData(candidate, Collections.<String>emptyList());
             if (version.isAvailable()) {
-                InstalledGcc gcc = new InstalledGcc("gcc");
-                if (i > 0) {
+                InstalledGcc gcc = new InstalledGcc("gcc" + " " + version.getVersion());
+                if (candidate != firstInPath) {
```

Comment:
This might work correctly now, but I'd rather we used `equals()` so that it keeps working even if we change the surrounding code later.